File: src/lang/javascript/prime-generator.js
function primesIn(n) {
	const vis = new Array(n + 1).fill(false);
	const primes = [];
	for (let i = 2; i <= n; i++) {
		if (vis[i]) continue;
		primes.push(i);
		for (let j = i * i; j <= n; j += i) vis[j] = true;
	}
	return primes;
}

File: src/lang/python/prime_generator.py
def primes_in(n: int) -> list[int]:
    primes = []
    vis = [False] * (n + 1)
    for i in range(2, n + 1):
        if vis[i]:
            continue
        primes.append(i)
        for j in range(i * i, n + 1, i):
            vis[j] = True
    return primes
